spring-cloud-app-broker-deployer/src/main/java/org/springframework/cloud/appbroker/deployer/GetApplicationRequest.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2002-2019 the original author or authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.appbroker.deployer;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.springframework.util.CollectionUtils;
+
+public class GetApplicationRequest {
+
+	private final String name;
+
+	private final Map<String, String> properties;
```

Comment:
Should this be a `Map<String, Object>`?

Reply:
We are using `Map<String, String>` in other files when referring to properties, e.g. `DeployApplicationRequest`, `GetServiceInstanceRequest`. We should keep it consistent.

Reply:
Agreed with @LittleBaiBai, we have properties as `Map<String, String>` since they are typically key-value and we have parameters as `Map<String, Object>` since they can be nested.

Reply:
Understood. I like consistency. Maybe this should be something we reconsider in 2.x? `String` is still limiting for storing numeric values (unless you store them as strings), regardless of needing to nest properties.